django/template/base.py

```diff
@@ -122,6 +122,32 @@ def __str__(self):
         return self.msg % tuple(force_text(p, errors='replace') for p in self.params)
 
 
+class UndefinedVariable:
+    """
+    This is the result of evaluating a template expression that
+    contains an undefined variable.
+
+    An instance of this class evaluates to False in a boolean
+    context. All instances are treated as being equal to each other.
+    """
+
+    def __init__(self, string_rep=""):
+        """
+        :param string_rep: The string value that this expression should
```

Comment:
We don't use JavaDoc style.
```
`string_repo` is the ...
```
would be fine. Maybe call it `var_repr` (usually variables aren't named after the type that they are).